docs/book/src/cronjob-tutorial/testdata/finalizer_example.go

```diff
@@ -68,23 +68,23 @@ func (r *CronJobReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		// The object is being deleted
 		if containsString(cronJob.ObjectMeta.Finalizers, myFinalizerName) {
 			// our finalizer is present, so lets handle any external dependency
-			if err := r.deleteExternalResources(cronJob); err != nil {
+			if err := r.deleteExternalResources(&cronJob); err != nil {
 				// if fail to delete the external dependency here, return with error
 				// so that it can be retried
 				return ctrl.Result{}, err
 			}
 
 			// remove our finalizer from the list and update it.
 			cronJob.ObjectMeta.Finalizers = removeString(cronJob.ObjectMeta.Finalizers, myFinalizerName)
-			if err := r.Update(context.Background(), cronJob); err != nil {
+			if err := r.Update(context.Background(), &cronJob); err != nil {
 				return ctrl.Result{}, err
 			}
 		}
-
-		return ctrl.Result{}, err
 	}
 
 	// rest of the reconciler code
+
+	return ctrl.Result{}, nil
```

Comment:
The comment `// rest of the reconciler code` is not expected to be empty like in this example, and it shouldn't run if the object is being deleted, so the `return` inside the `else` is needed. If you want to add the `return` also at the end is ok, but it was being included in the `// rest of the reconciler code`.